File: app/src/main/java/com/ngengs/android/popularmovies/apps/data/remote/MoviesList.java
package com.ngengs.android.popularmovies.apps.data.remote;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by ngengs on 6/15/2017.
 */

@SuppressWarnings({"unused", "DefaultFileTemplate"})
public class MoviesList {
    @SerializedName("page")
    private int page;
    @SerializedName("total_result")
    private int totalResult;
    @SerializedName("total_pages")
    private int totalPage;
    @SerializedName("results")
    private List<MoviesDetail> movies;

    //Error Data
    @SerializedName("status_code")
    private int statusCode;
    @SerializedName("status_message")
    private String statusMessage;

    public MoviesList(List<MoviesDetail> movies) {
        this.movies = movies;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getTotalResult() {
        return totalResult;
    }

    public void setTotalResult(int totalResult) {
        this.totalResult = totalResult;
    }

    public int getTotalPage() {
        return totalPage;
    }

    public void setTotalPage(int totalPage) {
        this.totalPage = totalPage;
    }

    public List<MoviesDetail> getMovies() {
        return movies;
    }

    public void setMovies(List<MoviesDetail> movies) {
        this.movies = movies;
    }

    public int getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }

    public String getStatusMessage() {
        return statusMessage;
    }

    public void setStatusMessage(String statusMessage) {
        this.statusMessage = statusMessage;
    }
}

File: app/src/main/java/com/ngengs/android/popularmovies/apps/data/remote/MoviesList.kt
package com.ngengs.android.popularmovies.apps.data.remote

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

/**
 * Created by rizky.kharisma on 10/11/22.
 * @ngengs
 */
@Parcelize
data class MoviesList(
    @SerializedName("page")
    val page: Int = 0,
    @SerializedName("total_result")
    val totalResult: Int = 0,
    @SerializedName("total_pages")
    val totalPage: Int = 0,
    @SerializedName("results")
    val movies: List<MoviesDetail> = mutableListOf(),

    // Error Data
    @SerializedName("status_code")
    val statusCode: Int? = null,
    @SerializedName("status_message")
    val statusMessage: String? = null
) : Parcelable